content/cloud-docs/api-docs/providers.mdx

```diff
@@ -254,89 +318,1138 @@ curl \
+| ------- | ---------------------------------------------------- | ---------------------------------------------------------- |
+| [200][] | [JSON API document][] (`type: "registry-providers"`) | The request was successful                                 |
+| [404][] | [JSON API error object][]                            | GPG key not found or user not authorized                   |
+
+### Query Parameters
+
+This endpoint supports pagination [with standard URL query parameters](/cloud-docs/api-docs/#query-parameters). Remember to percent-encode `[` as `%5B` and `]` as `%5D` if your tooling doesn't automatically encode URLs.
+
+| Parameter            | Description                                                                                                                                                           |
+| -------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `filter[field name]` | **Required.** Restricts results to those with the matching field name value. A valid value is `namespace` and can contain multiples (`filter[namespace]=org1, org2`). |
 
-### Sample Request (public provider)
+### Sample Request
 
 ```shell
 curl \
   --request GET \
+  --header "Authorization: Bearer $TOKEN" \
+  https://app.terraform.io/api/registry/private/v2/gpg-keys?filter[namespace]=hashicorp
+```
+
+### Sample Response
+
+```json
+{
+  "data": [
+    {
+      "type": "gpg-keys",
+      "id": "23",
+      "attributes": {
+        "ascii-armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINB...=txfz\n-----END PGP PUBLIC KEY BLOCK-----\n",
+        "created-at": "2022-02-11T19:16:59Z",
+        "key-id": "32966F3FB5AC1129",
+        "namespace": "hashicorp",
+        "source": "",
+        "source-url": null,
+        "trust-signature": "",
+        "updated-at": "2022-02-11T19:16:59Z"
+      },
+      "links": {
+        "self": "/v2/gpg-keys/23"
+      }
+    }
+  ],
+  "links": {
+    "first": "/v2/gpg-keys?filter%5Bnamespace%5D=hashicorp&page%5Bnumber%5D=1&page%5Bsize%5D=15",
+    "last": "/v2/gpg-keys?filter%5Bnamespace%5D=hashicorp&page%5Bnumber%5D=1&page%5Bsize%5D=15",
+    "next": null,
+    "prev": null
+  },
+  "meta": {
+    "pagination": {
+      "page-size": 15,
+      "current-page": 1,
+      "next-page": null,
+      "prev-page": null,
+      "total-pages": 1,
+      "total-count": 1
+    }
+  }
+}
+```
+
+## Add a GPG Key (Private Provider)
+
+`POST /api/registry/:registry_name/v2/gpg-keys`
+
+### Parameters
+
+| Parameter            | Description          |
+| -------------------- | -------------------- |
+| `:registry_name`     | Must be `private`.   |
+
+Uploads a GPG Key to a private registry scoped with a namespace. The response will provide a "key-id" which is needed to [Create a Provider Version](#create-a-provider-version) (see below).
```

Comment:
Just checking - should the parameter be in a code block? (if this isn't how we do it elsewhere, ignore me)

```suggestion
Uploads a GPG Key to a private registry scoped with a namespace. The response will provide a "key-id", which is required to [Create a Provider Version](#create-a-provider-version).
```

Reply:
The key path params further below need a code black (seen further beneath Sample Payload).

These params are referring to the path url :)